### ISSUE TITLE
Create API endpoint tasks/index.ts

### ISSUE THREAD

**Issue by kleneway** [PioneerSquareLabs/otto-playground#119]
Create the API endpoint tasks/index.ts that will API route for getting all tasks. Be sure to add the proper extension when creating the file. Here are the instructions:

## Issue Description

In this issue, we will create a new Next.js API endpoint for getting all tasks in the system. The endpoint will be a `GET` request to `/api/tasks`. The purpose of this endpoint is to fetch all tasks from the database and return them in a JSON format. The response should be an array of task objects, each containing the task's details.

### Requirements

1. Create a new file at `~/pages/api/tasks/index.ts` for the API endpoint.
2. Import the following dependencies at the top of the file:

   ```typescript
   import { NextApiRequest, NextApiResponse } from "next";
   import { Task } from "~/types";
   import { TaskSchema } from "~/types";
   import db from "~/server/db";
   ```

3. Define a TypeScript type for the expected output:

   ```typescript
   type GetTasksResponse = {
     tasks: Task[];
   };
   ```

4. Create an async function called `handler` with the following signature:

   ```typescript
   const handler = async (req: NextApiRequest, res: NextApiResponse<GetTasksResponse | Error>): Promise<void> => {
     // Implementation goes here
   };
   ```

5. Inside the `handler` function, check if the request method is `GET`. If not, return a `405 Method Not Allowed` status code with an error message:

   ```typescript
   if (req.method !== "GET") {
     res.status(405).json({ message: "Method Not Allowed" });
     return;
   }
   ```

6. Query the database to fetch all tasks using Prisma:

   ```typescript
   const tasks = await db.task.findMany();
   ```

7. Validate the fetched tasks using Zod:

   ```typescript
   const result = TaskSchema.array().safeParse(tasks);

   if (!result.success) {
     res.status(500).json({ message: "Error validating tasks data" });
     return;
   }
   ```

8. Return the validated tasks in the response with a `200 OK` status code:

   ```typescript
   res.status(200).json({ tasks: result.data });
   ```

9. Finally, export the `handler` function as the default export:

   ```typescript
   export default handler;
   ```

### Error Cases

- If the request method is not `GET`, return a `405 Method Not Allowed` status code with an error message.
- If there is an error validating the fetched tasks data, return a `500 Internal Server Error` status code with an error message.

### Summary

To summarize, this issue requires the creation of a new API endpoint at `~/pages/api/tasks/index.ts` that supports a `GET` request to fetch all tasks from the database. The endpoint should validate the fetched tasks using Zod and return them in a JSON format with a `200 OK` status code. The response should be an array of task objects, each containing the task's details.
